src/layers/kan.jl

```diff
@@ -0,0 +1,316 @@
+# end
+
+# function initialparameters(
+#         rng::AbstractRNG, kan::KANDense{spT, bT, order}) where {spT, bT, order}
+#     grid = __init_kan_grid(
+#         kan.in_dims, kan.out_dims, kan.num_grid_intervals, kan.grid_range)
+#     noises = (kan.init_noise(rng, kan.in_dims * kan.out_dims, size(grid, 2)) .- (1 // 2)) .*
+#              kan.scale_noise ./ kan.num_grid_intervals
+#     coefficients = __curve_to_coefficient(grid, noises, grid, Val(order))
+#     ps = (; coefficients)
+#     spT && (ps = merge(
+#         ps, (; scale_spline=kan.init_scale_spline(rng, kan.in_dims, kan.out_dims))))
+#     bT &&
+#         (ps = merge(ps, (; scale_base=kan.init_scale_base(rng, kan.in_dims, kan.out_dims))))
+#     return ps
+# end
+
+# function initialstates(::AbstractRNG, kan::KANDense{spT, bT, order}) where {spT, bT, order}
+#     grid = __init_kan_grid(
+#         kan.in_dims, kan.out_dims, kan.num_grid_intervals, kan.grid_range)
+#     mask = ones(eltype(grid), (kan.in_dims, kan.out_dims))
+#     st = (; grid, mask, weight_sharing=Colon(), lock_counter=0,
+#         lock_id=zeros(Int, kan.in_dims, kan.out_dims))
+#     !spT && (st = merge(
+#         st, (; scale_spline=kan.init_scale_spline(rng, kan.in_dims, kan.out_dims))))
+#     !bT &&
+#         (st = merge(st, (; scale_base=kan.init_scale_base(rng, kan.in_dims, kan.out_dims))))
+#     return st
+# end
+
+# function __init_kan_grid(
+#         in_dims, out_dims, num_grid_intervals, grid_range::Tuple{T1, T2}) where {T1, T2}
+#     T = promote_type(T1, T2)
+#     nsplines = in_dims * out_dims
+#     return ones(T, nsplines) .*
+#            LinRange(grid_range[1], grid_range[2], num_grid_intervals + 1)'
+# end
+
+# # @generated needed else Zygote is type unstable
+# @views @generated function (kan::KANDense{spT, bT, order})(
+#         x::AbstractMatrix, ps, st::NamedTuple) where {spT, bT, order}
+#     scale_expr = spT ? :(ps.scale_spline) : :(st.scale_spline)
+#     base_expr = bT ? :(ps.scale_base) : :(st.scale_base)
+#     return quote
+#         B = size(x, 2)
+#         scale_spline = $(scale_expr)
+#         scale_base = $(base_expr)
+#         x_ = reshape(x, :, 1, B)
+#         preacts = x_ .* __ones_like(x, (1, kan.out_dims, 1))                  # I x O x B
+#         y = __coefficient_to_curve(reshape(preacts, :, B), st.grid[st.weight_sharing, :],
+#             ps.coefficients[st.weight_sharing, :], $(Val(order)))             # (I x O) x B
+#         postspline = reshape(y, kan.in_dims, kan.out_dims, B)
+#         mask = CRC.ignore_derivatives(st.mask)
+#         postacts = @. (scale_base * kan.activation(x_) + scale_spline * postspline) * mask
+#         res = dropdims(sum(postacts; dims=1); dims=1)                         # O x B
+#         return res, st
+#     end
+# end
+
+# utilities for splines
+## TODO: Upstream to NNlib
+@inline silu(x) = x * sigmoid(x)  # Not present in NNlib
+@inline silu_fast(x) = x * sigmoid_fast(x)
```

Comment:
@avik-pal `silu` is the same as `swish` which is defined in NNlib.
Maybe you can upstream your `silu_fast` implementation as `NNlib.swish_fast`.

Reply:
Okay the swish in NNlib is same as the silu_fast defined here, I will delete these then